python/utils.py

```diff
@@ -499,8 +500,13 @@ def reloadPlugin(packageName):
     startPlugin(packageName)
 
 
-def showPluginHelp(packageName=None, filename="index", section=""):
-    """ show a help in the user's html browser. The help file should be named index-ll_CC.html or index-ll.html"""
+def showPluginHelp(packageName: str = None, filename: str = "index", section: str = ""):
```

Comment:
It is not completely clear to me, if python type hints are accepted.
Looking at the code base, they are not used widely (I found one single occurrence in the BatchAlgorithmDialog). The INSTALL.md mentions minimum Python version to be 3.3 but if I understand it correctly, this was introduced in Python 3.5.

However:
- I would be absolutely in favor of adding this as a step towards modern Python
- A bigger dependency version bump has been discussed
- Most our target distributions have Python >= 3.5
- The occurrence in the BatchAlgorithmDialog has survived the last 2 years

Ping @nyalldawson because you introduced the one in [the batch dialog 1.5 years ago](https://github.com/qgis/QGIS/commit/e0093e1d1135ced14865458da493de4f6e75ff7c#diff-9fe557e6a4a817c6c22860b91ad575344bdf135dc7d0437b6ff5bee9cbc9d6b7R60) and have been involved with the dependency bump discussion.

Reply:
Given the pending min version increment to 3.6 and that there's other code which uses typehints, I think we should allow (and encourage!!) this